environment/src/main/java/com/sequenceiq/environment/environment/flow/deletion/handler/ResourceEncryptionDeleteHandler.java

```diff
@@ -57,10 +92,63 @@ public void accept(Event<EnvironmentDeletionDto> environmentDtoEvent) {
         }
     }
 
-    private void deleteEncryptionResources(Environment environment) {
-        /*
-        TODO: Delete the Disk Encryption Set if it exists.
-        */
+    private void deleteEncryptionResources(EnvironmentDto environmentDto, Environment environment, String diskEncryptionSetUrl) {
+        if (environmentEncryptionService.deleteEncryptionSet(diskEncryptionSetUrl)) {
+            LOGGER.info("Successfully deleted diskEncryptionSet");
+        } else {
+            LOGGER.error("Failed to delete diskEncryptionSet");
+        }
+        if (!isSingleResourceGroup(environmentDto)) {
+            String resourceGroupName = getResourceGroupNameFromDES(diskEncryptionSetUrl);
+            if (StringUtils.isNotEmpty(resourceGroupName)) {
+                deleteAzureResourceGroup(resourceGroupName, environment);
+            } else {
+                LOGGER.error("Failed to parse Encryption parameters resource group");
+            }
+        }
```

Comment:
The DES + RG deletion shall be considered a single operation at the `cloud-api` level. This is an implementation detail. Both procedures have to be executed in the actual Azure impementation class, i.e. in `AzureEncryptionResources` or some service called therein.